kfs-core/src/main/java/edu/arizona/kfs/gl/businessobject/EntryBo.java

```diff
@@ -0,0 +1,397 @@
+        return projectCode;
+    }
+
+    public void setProjectCode(String projectCode) {
+        this.projectCode = projectCode;
+    }
+
+    public String getOrganizationReferenceId() {
+        return organizationReferenceId;
+    }
+
+    public void setOrganizationReferenceId(String organizationReferenceId) {
+        this.organizationReferenceId = organizationReferenceId;
+    }
+
+    public String getReferenceFinancialDocumentTypeCode() {
+        return referenceFinancialDocumentTypeCode;
+    }
+
+    public void setReferenceFinancialDocumentTypeCode(String referenceFinancialDocumentTypeCode) {
+        this.referenceFinancialDocumentTypeCode = referenceFinancialDocumentTypeCode;
+    }
+
+    public String getReferenceFinancialSystemOriginationCode() {
+        return referenceFinancialSystemOriginationCode;
+    }
+
+    public void setReferenceFinancialSystemOriginationCode(String referenceFinancialSystemOriginationCode) {
+        this.referenceFinancialSystemOriginationCode = referenceFinancialSystemOriginationCode;
+    }
+
+    public String getReferenceFinancialDocumentNumber() {
+        return referenceFinancialDocumentNumber;
+    }
+
+    public void setReferenceFinancialDocumentNumber(String referenceFinancialDocumentNumber) {
+        this.referenceFinancialDocumentNumber = referenceFinancialDocumentNumber;
+    }
+
+    public Date getFinancialDocumentReversalDate() {
+        return financialDocumentReversalDate;
+    }
+
+    public void setFinancialDocumentReversalDate(Date financialDocumentReversalDate) {
+        this.financialDocumentReversalDate = financialDocumentReversalDate;
+    }
+
+    public String getTransactionEncumbranceUpdateCode() {
+        return transactionEncumbranceUpdateCode;
+    }
+
+    public void setTransactionEncumbranceUpdateCode(String transactionEncumbranceUpdateCode) {
+        this.transactionEncumbranceUpdateCode = transactionEncumbranceUpdateCode;
+    }
+
+    public Date getTransactionPostingDate() {
+        return transactionPostingDate;
+    }
+
+    public void setTransactionPostingDate(Date transactionPostingDate) {
+        this.transactionPostingDate = transactionPostingDate;
+    }
+
+    public Timestamp getTransactionDateTimeStamp() {
+        return transactionDateTimeStamp;
+    }
+
+    public void setTransactionDateTimeStamp(Timestamp transactionDateTimeStamp) {
+        this.transactionDateTimeStamp = transactionDateTimeStamp;
+    }
+
+    public String getGecDocumentNumber() {
+        if (gecDocumentNumber == null) {
+            gecDocumentNumber = generateGecDocumentNumber();
+        }
+        return gecDocumentNumber;
+    }
+
+    public void setGecDocumentNumber(String gecDocumentNumber) {
+        this.gecDocumentNumber = gecDocumentNumber;
+    }
+
+    public String getEntryId() {
+        if (entryId == null) {
+            entryId = generateEntryId();
+        }
+        return entryId;
+    }
+
+    private String generateGecDocumentNumber() {

```

Comment:
Is this really generating anything new here? I.e., is this methods name a misnomer? Looks to me like your returning accoutline doc number if the conditions are right.